akka-cluster/src/multi-jvm/scala/akka/cluster/SplitBrainQuarantineSpec.scala

```diff
@@ -0,0 +1,116 @@
+  // partition is resolved and the two split brain halves reconnects
+  // Not intended to be a part of the final PR
+
+  "Cluster node downed by other" must {
+
+    if (!ArterySettings(system.settings.config.getConfig("akka.remote.artery")).Enabled) {
+      // this feature only works in Artery, because classic remoting will not accept connections from
+      // a quarantined node, and that is too high risk of introducing regressions if changing that
+      pending
+    }
+
+    "join cluster" taggedAs LongRunningTest in {
+      awaitClusterUp(first, second, third, fourth)
+      enterBarrier("after-1")
+    }
+
+    "split brain" taggedAs LongRunningTest in {
+      runOn(first) {
+        testConductor.blackhole(first, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(first, fourth, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(second, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(second, fourth, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("blackhole")
+      system.log.info("cluster split into [JVM-1, JVM-2] and [JVM-3, JVM-4] with blackhole")
+
+      within(15.seconds) {
+        runOn(first) {
+          cluster.down(third)
+          cluster.down(fourth)
+        }
+        runOn(first, second) {
+          awaitAssert {
+            cluster.state.members.collect { case m if m.status == MemberStatus.Up => m.address } should ===(
+              Set(address(first), address(second)))
+          }
+          system.log.info("JVM-3 and JVM-4 downed from JVM-1")
+        }
+
+        runOn(third) {
+          cluster.down(first)
+          cluster.down(second)
+        }
+        runOn(third, fourth) {
+          awaitAssert {
+            cluster.state.members.collect { case m if m.status == MemberStatus.Up => m.address } should ===(
+              Set(address(third), address(fourth)))
+          }
+          system.log.info("JVM-1 and JVM-2 downed from JVM-3")
+        }
+      }
+      enterBarrier("brain-split")
+      system.log.info("Brain split, waiting 15s")
+      Thread.sleep(15000)
+
+      runOn(first) {
+        system.log.info("unblackholing cluster")
+        testConductor.passThrough(first, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(first, fourth, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(second, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(second, fourth, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("unblackholed")
+      // FIXME give it some time so we see what happens now before test infra killing systems
+      Thread.sleep(30000)
+      enterBarrier("after-pass-through")
+      cluster.isTerminated should ===(true) // all nodes should have shut down?
```

Comment:
I looked at the logs for this one also. It must send some real messages to be relevant.
```
runOn(first, second) {
        system.actorSelection(RootActorPath(third) / "user").tell(Identify(None), ActorRef.noSender)
        system.actorSelection(RootActorPath(fourth) / "user").tell(Identify(None), ActorRef.noSender)
      }
      runOn(third, fourth) {
        system.actorSelection(RootActorPath(first) / "user").tell(Identify(None), ActorRef.noSender)
        system.actorSelection(RootActorPath(second) / "user").tell(Identify(None), ActorRef.noSender)
      }
      Thread.sleep(5000)
      runOn(first) {
        system.actorSelection(RootActorPath(third) / "user").tell(Identify(None), ActorRef.noSender)
        system.actorSelection(RootActorPath(fourth) / "user").tell(Identify(None), ActorRef.noSender)
      }
      runOn(third, fourth) {
        system.actorSelection(RootActorPath(first) / "user").tell(Identify(None), ActorRef.noSender)
        system.actorSelection(RootActorPath(second) / "user").tell(Identify(None), ActorRef.noSender)
      }
      Thread.sleep(5000)
      enterBarrier("after-pass-through")
      cluster.isTerminated should ===(false)
```

Note that it should not trigger shutdown of other side.

This is what happens:
```
[JVM-1] Quarantine piercing attempt with message [ActorSelectionMessage(akka.actor.Identify)] to [Actor[akka://SplitBrainQuarantine@third/]]

[JVM-1] Dropping message [akka.remote.artery.Quarantined] from [akka://SplitBrainQuarantine@third#-4845019687174480295] because the system is quarantined
```

That is good. It doesn't trigger the ThisActorSystemQuarantinedEvent because the Quarantined is dropped before that. If it would have published ThisActorSystemQuarantinedEvent then SBR would anyway not act on that because it's not from a member.

Reply:
Ok, good, I guess I had bad luck when thinking about what should happen once the partition goes away and thought about it from the aspect that it is an old member downed that turns out to be a zombie, but a quarantined message from a random node not in the cluster is more what it is.